pkg/dockerutil/dockerutils.go

```diff
@@ -839,6 +841,35 @@ func CheckDockerVersion(versionConstraint string) error {
 		}
 		return fmt.Errorf("%s", msgs)
 	}
+
+	// Check Docker API version
+	currentAPIVersion, err := GetDockerAPIVersion()
+	if err != nil {
+		return fmt.Errorf("unable to get Docker API version: %v", err)
+	}
+	apiVersion, err := semver.NewVersion(currentAPIVersion)
+	if err != nil {
+		return err
+	}
+
+	apiConstraint, err := semver.NewConstraint(requiredDockerAPIVersion)
+	if err != nil {
+		return err
+	}
+
+	apiMatch, apiErrs := apiConstraint.Validate(apiVersion)
+	if !apiMatch {
+		if len(apiErrs) <= 1 {
+			return apiErrs[0]
+		}
+
+		apiMsgs := "\n"
+		for _, err := range apiErrs {
+			apiMsgs = fmt.Sprint(apiMsgs, err, "\n")
+		}
+		return fmt.Errorf("%s", apiMsgs)
+	}
+
```

Comment:
This can be replaced by:

```go
import (
	...
	dockerVersions "github.com/docker/docker/api/types/versions"
	...
)

...

dockerAPIVersion, err := GetDockerAPIVersion()
if err != nil {
	return fmt.Errorf("unable to get Docker API version: %v", err)
}
if !dockerVersions.GreaterThanOrEqualTo(dockerAPIVersion, minimalDockerAPIVersion) {
	return fmt.Errorf("You %s doesn't match minimum Docker API version %s", dockerAPIVersion, minimalDockerAPIVersion)
}
return nil
```